editor/src/messages/portfolio/portfolio_message_handler.rs

```diff
@@ -171,14 +171,27 @@ impl MessageHandler<PortfolioMessage, (&InputPreprocessorMessageHandler, &Prefer
 				};
 
 				let copy_val = |buffer: &mut Vec<CopyBufferEntry>| {
-					for layer_path in active_document
+					let binding = active_document
 						.metadata()
-						.shallowest_unique_layers(active_document.selected_nodes.selected_layers(active_document.metadata()))
-					{
-						let Some(layer) = layer_path.last().copied() else {
-							continue;
-						};
+						.shallowest_unique_layers(active_document.selected_nodes.selected_layers(active_document.metadata()));
+
+					let mut get_last_elements: Vec<_> = binding.iter().map(|x| x.last().expect("empty path")).collect();
+
+					let selected_to_vector = active_document
+						.selected_nodes
+						.selected_layers(active_document.metadata())
+						.filter(|_x| true)
+						.map(|x| x.clone())
+						.collect::<Vec<_>>();
```

Comment:
I don't see why it is necessary to collect this and reallocate then sort the other list such that it is in the same order as this one. A more efficient method would probably be just to do something like`active_document.metadata.all_layers().filter(|layer| get_last_elements.contains(&layer)).collect()` and then not do sorting.

Reply:
Yes I understand